specs/interop/liquidity_migration.md

```diff
@@ -0,0 +1,358 @@
+# Liquidity Migration
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+
+- [Overview](#overview)
+- [Implementation changes](#implementation-changes)
+  - [External functions](#external-functions)
+    - [`convertToSuper`](#converttosuper)
+    - [`convertFromSuper`](#convertfromsuper)
+  - [Events](#events)
+    - [`ConvertedToSuper`](#convertedtosuper)
+    - [`ConvertedFromSuper`](#convertedfromsuper)
+  - [Internal Functions](#internal-functions)
+    - [`checkPair`](#checkpair)
+- [Diagram](#diagram)
+- [Invariants](#invariants)
+- [Considerations](#considerations)
+- [Appendix](#appendix)
+  - [Access Control](#access-control)
+  - [Decimal management](#decimal-management)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Overview
+
+The `OptimismMintableERC20` tokens and `L2StandardToken`s (_legacy tokens_),
+which correspond to locked liquidity in L1, are incompatible with interop.
+Legacy token owners must convert into a `SuperchainERC20` representation, as defined in the [standard](token-bridging.md),
+to move across the Superchain.
+
+The conversion method will use the `L2StandardBridge` mint/burn rights
+over the legacy tokens to allow easy migration to and from the
+corresponding `SuperchainERC20`.
+
+## Implementation changes
+
+The `L2StandardBridge` will be extended to include the following functions and events:
```

Comment:
Perhaps you could make a `### L2StandardBridge` section and a `### OptimismMintableERC20TokenFactory` section so that we have headers for the specific contracts getting changes

Reply:
This is good that it lists the specific changes for the `L2StandardBridge`, it would be good to note the specific changes to the factory, ie the new mapping for deployed contracts

Reply:
I see the following pseudocode below:

```solidity
  address _legacyRemoteToken = factory.deployments(_legacyAddr);
```

Reply:
That's a great idea, will do it <3

Reply:
Done